src/mesh.c

```
@@ -6,381 +6,325 @@
+int mesh_update_routing(uint16_t source, uint16_t from_node, int16_t rssi, uint8_t band) {
+    if (source == 0 || source == mesh_state.our_node_id) return -1;
+
+    int safe_band = (band < 3) ? (int)band : 0; /* clamp to valid range for neighbor.rssi[3] */
 
-    /* If we received this directly (from_node == 0), update neighbor info */
-    if (from_node == 0)
-    {
+    if (from_node == 0) {
+        /* Direct link: heard directly from the source */
         mesh_add_neighbor(source, band, rssi);
+        mesh_add_route(source, source, 1, safe_band ? -90 : rssi, band);
```

Clang (14.0.6)

[warning] narrowing conversion from 'int' to signed type 'int8_t' (aka 'signed char') is implementation-defined

(bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions)